tests/System.Text.Primitives.Tests/ParserTests.cs

```diff
@@ -4,472 +4,983 @@
 
 namespace System.Text.Primitives.Tests
 {
-    public class ParserTests
+	public class ParserTests
     {
-        private byte[] UtfEncode(string s)
+		private byte[] UtfEncode(string s)
         {
             return Encoding.UTF8.GetBytes(s);
         }
-        
-        #region ulong
-        [Theory]
-        [InlineData("1728", true, 0, 1728, 4)]
-        [InlineData("blahblahh1751110", true, 9, 1751110, 7)]
-        [InlineData("987abcdefg", true, 0, 987, 3)]
-        [InlineData("The biggest ulong is 9223372036854775808.", true, 21, 9223372036854775808, 19)]
+        private byte[] UtfEncode(string s, bool utf16)
+        {
+            if (utf16)
+                return Encoding.Unicode.GetBytes(s);
+            else
+                return UtfEncode(s);
+        }
+
+		#region byte
+
+		[Theory]
+        [InlineData("55", true, 0, 55, 2)]
```

Comment:
If all of this `InlineData` is going to be the same for multiple methods, I would recommend putting it into a member function `X` and then using [MemberData(nameof(X))] instead of duplicating it into each method's attributes. It looks like you can share it for all 4 Utf8/Utf16 array/pointer versions of these tests.